llvm/lib/Transforms/InstCombine/InstCombineAndOrXor.cpp

```diff
@@ -4250,10 +4250,11 @@ static Instruction *canonicalizeAbs(BinaryOperator &Xor,
     // xor (add A, Op1), Op1  ; add -1 and flip bits if negative
     // --> (A < 0) ? -A : A
     Value *IsNeg = Builder.CreateIsNeg(A);
-    // Copy the nuw/nsw flags from the add to the negate.
+    // Copy the nsw flags from the add to the negate.
     auto *Add = cast<BinaryOperator>(Op0);
-    Value *NegA = Builder.CreateNeg(A, "", Add->hasNoUnsignedWrap(),
-                                   Add->hasNoSignedWrap());
+    Value *NegA = Add->hasNoUnsignedWrap()
```

Comment:
This change fixes a regression caused by not propagating nuw flags.

Reply:
Why is dropping the `nuw` APIs from neg related to this fixing the bug?Should this be independent commits?

Reply:
I don't know if it is OK to drop these changes and leave the regression.

Reply:
Oh I see what you mean. I misunderstood and  thought you meant this patch was really all about this little fix here.